content/docs/reference/design-docs/signaling/speed-limits.en.md

```diff
@@ -0,0 +1,78 @@
+---
+title: "Speed limits"
+linkTitle: "Speed limits"
+weight: 30
+description: "Describes how speed limits work"
+---
+
+## Description
+
+Railway infrastructure has a surprising variety of speed limits:
+ - some are known by the driver, and not announced at all
+ - some are announced by fixed signs regardless of where the train goes
+ - some are announced by fixed signs, depending on where the train path goes
+ - some are announced by dynamic signals regardless of where the train goes
+ - some are announced by dynamic signals, depending on where the train path goes
+
+## Data model
+
+```yaml
+{
+    # unique speed limit identifier
+    "id": "...",
+
+    # A list of routes the speed limit is enforced on. When empty or missing,
+    # the speed limit is enforced regardless of the route. If a signal announces
+    # this speed section on a given route, it is automatically added to this list.
+    "on_routes": ["${ROUTE_A}", "${ROUTE_B}"]
+
+    # speed limit in meters per second
+    "speed_limit": 30,
+    # a map from train tag to speed limit override
+    "speed_limit_by_tag": {"freight": 20},
+
+    "track_ranges": [{"track": "${TRACK_SECTION}", "begin": 0, "end": 42, "applicable_directions": "START_TO_STOP"}],
+}
+```
+
+## Design considerations
+
+### Where to put the speed limit
+
+When a speed limit is announced by dynamic signaling, we may be in a position where speed limit information is duplicated:
+ - once in the signal itself
+ - once in the speed limit
+
+For example, we may have a `RR30` signal before a 30km/h speed limit. This is not great, and quite error prone.
+
+There are three options:
+1) accept it, and find a way to validate that both ends match
+2) deduce the signal constraint from the speed limit
+3) deduce the speed limit from the signal
```

Comment:
advantages :
- easier to edit
- less verification needed
- closer to reality
- less duplication
- easier to import (if only available with signals)
- less speed sections implies better performances for generating layers

disadvantages : 
- can we handle all the cases ? 
- harder to simulate (need train reactions)
- more difficult to create a layer of this speed limits

Reply:
I would like to clarify that option 3 was not about the bounds of the speed **sections**, just deducing the value of the speed **limit**

Reply:
> can we handle all the cases ?

Apparently not, Ralen 30/60 can have an execution point different from both the first switch and the rappel signal.

Reply:
After more discussion, I've integrated a description of this idea into the design document (as "Inlining speed limit definitions into signals")